docs/guides/intro-to-graphql.md

```diff
@@ -1,9 +1,407 @@
+
+Executing mutations is the same, however we need to pass variables along with the mutation:
+
+```javascript
+const ADD_USER = `
+mutation AddUser($name: String!) {
+  createUser(data: { name: $name }) {
+    name
+    id
+  }
+}`;
+```
+
+The key for mutations is still `query`:
+
+```javascript
+fetch('/admin/api', {
+  method: 'POST',
+  headers: {
+    'Content-Type': 'application/json',
+  },
+  body: {
+    query: ADD_USER,
+    variables: { name: 'Mike' },
+  },
+}).then(result => result.json());
+```
+
+**Note: ** Queries via the API will enforce [Access Control](https://v5.keystonejs.com/api/access-control).
+
+## Executing Queries and Mutations on the Server
+
+In addition to executing queries via the API, you can execute queries and mutations on the server using the `keystone.executeQuery` method.
+
+**Note: ** No access control checks are run when executing queries on the server. Any queries or mutations that checked for `context.req` in the resolver may also return different results as the `req` object is set to `{}`.
+
+See: [Keystone executeQuery()](https://v5.keystonejs.com/keystone-alpha/keystone/#executequeryquerystring-config)
+
+## Search and Filtering
+
+When executing queries and mutations there are a number of ways you can filter, limit and sort items. These include:
+
+- `where`
+- `search`
+- `skip`
+- `first`
+- `orderby`
+
+#### `where`
+
+Limit results to items matching the where clause. Where clauses are used to query fields in a keystone list before retrieving data.
+
+The options available in a where clause depend on the field types.
+
+```gql
+query {
+  allUsers (where: { id_starts_with_i: 'A'} ) {
+    id
+  }
+}
+```
+
+##### Relationship `where` filters
+
+- `{relatedList}_every`: whereInput
+- `{relatedList}_some`: whereInput
+- `{relatedList}_none`: whereInput
+- `{relatedList}_is_null`: Boolean
+
+##### String `where` filters
+
+- `{Field}:` String
+- `{Field}_not`: String
+- `{Field}_contains`: String
+- `{Field}_not_contains`: String
+- `{Field}_starts_with`: String
+- `{Field}_not_starts_with`: String
+- `{Field}_ends_with`: String
+- `{Field}_not_ends_with`: String
+- `{Field}_i`: String
+- `{Field}_not_i`: String
+- `{Field}_contains_i`: String
+- `{Field}_not_contains_i`: String
+- `{Field}_starts_with_i`: String
+- `{Field}_not_starts_with_i`: String
+- `{Field}_ends_with_i`: String
+- `{Field}_not_ends_with_i`: String
+- `{Field}_in`: [String]
+- `{Field}_not_in`: [String]
+
+##### ID `where` filters
+
+- `{Field}`: ID
+- `{Field}_not`: ID
+- `{Field}_in`: [ID!]
+- `{Field}_not_in`: [ID!]
+
+##### Integer `where` filters
+
+- `{Field}: Int`
+- `{Field}_not`: Int
+- `{Field}_lt`: Int
+- `{Field}_lte`: Int
+- `{Field}_gt`: Int
+- `{Field}_gte`: Int
+- `{Field}_in`: [Int]
+- `{Field}_not_in`: [Int]
+
+#### Operators
+
+You can combine multiple where clauses with `AND` or `OR` operators.
+
+- `AND`: [whereInput]
+- `OR`: [whereInput]
```

Comment:
This would benefit from an example:

```gql
query {
  allUsers (where: {
    OR: [
      { name_starts_with_i: 'A' },
      { email_starts_with_i: 'A' },
    ]
  } ) {
    id
  }
}
```

Reply:
Yeah I thought the same. Didn't want this to get too big but I guess if you thought this too maybe we should add it.